README.md

```diff
@@ -1 +1,166 @@
-# dask-pytorch
\ No newline at end of file
+# dask-pytorch
+
+<!-- ![GitHub Actions](https://github.com/saturncloud/dask-pytorch/workflows/GitHub%20Actions/badge.svg) [![PyPI Version](https://img.shields.io/pypi/v/prefect-saturn.svg)](https://pypi.org/project/prefect-saturn) -->
+
+`dask-pytorch` is a Python package that makes it easy to train PyTorch models on dask clusters using distributed data paralllel.  The intended scope of the project is
+- bootstrapping PyTorch workers on top of a Dask cluster
+- Implementations of common PyTorch datasets on distributed data stores (like S3)
+- mechanisms for tracking and logging intermediate results, training statistics, and checkpoints.
+
+## Typical non-dask workflow
+
+A typical example of non-dask PyTorch usage is as follows:
+
+### Loading Data
+Create an dataset (ImageFolder), and wrap it in a DataLoader
+
+```python
+transform = transforms.Compose([
+    transforms.Resize(256),
+    transforms.CenterCrop(250),
+    transforms.ToTensor()
+])
+
+whole_dataset = ImageFolder(path, transform=transform)
+
+batch_size = 100
+num_workers = 64
+indices = list(range(len(data)))
+np.random.shuffle(indices)
+train_idx = indices[:num]
+test_idx = indices[num:num+num]
+
+train_sampler = SubsetRandomSampler(train_idx)
+train_loader = DataLoader(data, sampler=train_sampler, batch_size=batch_size, num_workers=num_workers)
+```
+
+### Training a Model
+Loop over the dataset, and train the model by stepping the optimizer
+
+```python
+device = torch.device(0)
+net = models.resnet18(pretrained=False)
+model = net.to(device)
+device_ids = [0]
+
+criterion = nn.CrossEntropyLoss().cuda()
+lr = 0.001
+optimizer = optim.SGD(model.parameters(), lr=lr, momentum=0.9)
+count = 0
+for epoch in range(n_epochs):
+    model.train()  # Set model to training mode
+    for inputs, labels in train_loader:
+        dt = datetime.datetime.now().isoformat()
+        inputs = inputs.to(device)
+        labels = labels.to(device)
+        outputs = model(inputs)
+        _, preds = torch.max(outputs, 1)
+        loss = criterion(outputs, labels)
+
+        # zero the parameter gradients
+        optimizer.zero_grad()
+        loss.backward()
+        optimizer.step()
+        count += 1
+```
+
+## Now on Dask
+
+with dask_pytorch and pytorch distributed data parallel, we can train on multiple workers as follows:
+
+### Loading Data
+Load the dataset from S3, and explicitly set the multiprocessing context (Dask defaults to spawn, but pytorch is generally configured to use fork)
+
+```python
+from dask_pytorch.data import BOTOS3ImageFolder
```

Comment:
Wondering if we want to name this module `BOTOS3ImageFolder` or `BotoS3ImageFolder`.